src/Microsoft.AspNet.Hosting/WebHostBuilder.cs

```diff
@@ -0,0 +1,184 @@
+// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNet.Builder;
+using Microsoft.AspNet.Hosting.Builder;
+using Microsoft.AspNet.Hosting.Internal;
+using Microsoft.AspNet.Hosting.Server;
+using Microsoft.AspNet.Hosting.Startup;
+using Microsoft.Framework.ConfigurationModel;
+using Microsoft.Framework.DependencyInjection;
+using Microsoft.Framework.Logging;
+using Microsoft.Framework.Runtime;
+using Microsoft.Framework.Runtime.Infrastructure;
+
+namespace Microsoft.AspNet.Hosting
+{
+    public class WebHostBuilder
+    {
+        public const string EnvironmentKey = "ASPNET_ENV";
+
+        private readonly IServiceProvider _fallbackServices;
+        private readonly IHostingEnvironment _hostingEnvironment;
+        private readonly ILoggerFactory _loggerFactory;
+
+        private Action<IServiceCollection> _configureServices;
+        private IConfiguration _config;
+
+        // Only one of these should be set
+        private StartupMethods _startup;
+        private Type _startupType;
+        private string _startupAssemblyName;
+
+        // Only one of these should be set
+        private string _serverFactoryLocation;
+        private IServerFactory _serverFactory;
+
+        public WebHostBuilder() : this(null) { }
+
+        public WebHostBuilder(IServiceProvider fallbackServices)
```

Comment:
So today after nuking the default ctor(), we are calling GetRequiredService on the ISP, so things go kerboom if you try to pass in any arbitrary ISP...  What behavior do we want here?  The tests need to either pass in the old CallContextServiceLocator, or we can fix the import to allow this?